### ISSUE TITLE
[Improvement] Suggesting standarization of List-KR-Script

### ISSUE THREAD

**Issue by seia-soto** [List-KR/List-KR-Script#15]
I am suggesting the change to current repository to automatically form the structure using snippets or prebuilt ecosystem rather than keeping the current way that manually updating and forming resources.

Will you mind it?

**Comment by piquark6046** [comment]
An userscript manager can download a userscript like List-KR Script automatically.
And, how can maintainers of List-KR Script is become one of scriptlets?

**Comment by seia-soto** [comment]
Ah yes, as you said, clients are ready to use the script.

However, about the developer experience, I think it's better to use package managers to have the development cycle. About the use of word `scriptlet`, it's my misuse. You can replace scriptlet with `script` or something similar.

The development ecosystem that I am thinking can automatically

1. update the version
2. merge the rules from folders
3. reform the script structure to be applied to the targeted site fast as possible

**Comment by piquark6046** [comment]
Did you mean that List-KR contains predefined script with parameters that can be converted to `#%#` rule without `#%#//scriptlet()` or `##+js()` (pure-javascript rule) by a FilterComplier for an end-user?

**Comment by seia-soto** [comment]
Ah, it's impossible but I think it would be nice if we have a toolchain something can postprocess our scripts and merge it into an output script that can automatically inject to sites that we want.

For example, assume that the directory with files contains following content:

```yaml
sitesApplied: example.domain.tld
utilsRequired: LSKLib...
description: ...
script: ... (actual script goes here)
```

**Comment by piquark6046** [comment]
It seems that it is nice.
However, a 3rd-party filter can **not** inject a javascript rule (e.g. `#%#var isTrue == true`) into a website on AdGuard registered filters platform.

https://github.com/AdguardTeam/FiltersRegistry/#filters-metadata
https://github.com/AdguardTeam/FiltersCompiler/tree/master/src/main/utils/trust-levels

**Comment by seia-soto** [comment]
Yes, so we need to make workaround independently from List-KR, extending the current structure.

What I want is a command that composes all rules we defined locally into one output.

Not manually editing the `user.js` file.

I'll make an example when I have free time.

**Comment by seia-soto** [comment]
Workarounds are on going in here: https://github.com/seia-soto/List-KR-Script/tree/showcase/structurable-scripts
